File: Source/Mobile/PVScan.Mobile/Models/Barcode.cs
using System;
using System.Collections.Generic;
using System.Text;
using ZXing;

namespace PVScan.Mobile.Models
{
    public class Barcode
    {
        public int Id { get; set; }
        public BarcodeFormat Format { get; set; }
        public string Text { get; set; }
        public Coordinate ScanLocation { get; set; }
        public DateTime ScanTime { get; set; }
        public bool Favorite { get; set; }
        public string GUID { get; set; }
    }
}

File: Source/Web/Frontend/app/src/models/Barcode.ts
import Coordinate from "./Coordinate";

export default interface Barcode {
    Id?: number | undefined;
    BarcodeFormat: number; // Later convert to something more mnemonic
    Text: string;
    ScanLocation: Coordinate;
    ScanTime: Date;
    Favorite: boolean;
    Hash: string;
    GUID: string;
    LastUpdateTime: Date;
}
